### DISCUSSION TITLE
Suggestion: Make the base URL configurable and prefix optional

### DISCUSSION THREAD

**Discussion by maxacarvalho** [ash-jc-allen/short-url#114] [Ideas]
Hello,

Thanks for your solution, it's very nice.

I'd like to make a suggestion, to let the application users to choose the base URL and make the prefix optional.

At the moment, I can see that the `default_sort_url` is set to `config('app.url').'/'.$this->prefix().'/'.$this->urlKey`, which makes impossible to have a domain different than the application one for the shortened URL.

Use case:

Let's say I have a full domain app that's https://full-domain-app.com, but when shortening the URL, I want to use `https://fda.is`, so the URL is even shorter. Let's say I also want to remove the `prefix` altogether.

Something like this could do the trick:
```php
...
    protected function insertShortURLIntoDatabase(): ShortURL
    {
        $baseUrl = $this->baseUrl ?? config('app.url');

        $prefix = $this->prefix() ? "{$this->prefix()}/" : '';

        return ShortURL::create([
            'destination_url'                => $this->destinationUrl,
            'default_short_url'              => "{$baseUrl}/{$prefix}".$this->urlKey,
...
```

I could create a PR for that, but I think it's better to understand if that's something you would like to add to your package.

Thanks!

**Comment by ash-jc-allen** [comment]
Hey @maxacarvalho! I 100% agree that I think that would be a really useful feature. I've actually been working on a project myself that would probably benefit from this type of functionality in the future.

I'm more than happy to accept a PR for this functionality if you fancy making one 😄

**Comment by goellner** [comment]
would really really love this feature

**Comment by ash-jc-allen** [comment]
These changes should all be available from v7.0.0 onwards 😄

**Comment by goellner** [comment]
Is there any documentation available on how to use this feature? Wanted to use a custom domain for the short url e.g. tinylink.co/asdf 

I saw that the PR https://github.com/ash-jc-allen/short-url/pull/121 should have added the functionality for that, but not sure how to set the custom domain. i have a route setup already, but don't know how to set the domain before its generated. 

The route setup I currently have: 

```
Route::domain('http://short.test')->group(function () {
    Route::get('/{shortURLKey}', ShortURLController::class);
});
```

this is the output of my `php artisan route:list`: 
```
GET|HEAD        short.test/{shortURLKey} ...... AshAllenDesign\ShortURL › ShortURLController
``` 

The routing works, but the `default_short_url` returns the `config('app.url')` as a base, how can I change this?

**Reply by ash-jc-allen** [reply]
Hey @goellner!

Sorry, my previous comment wasn't great and made it sound like v7.0.0 included the functionality for custom domains and optional prefixes. But it only actually included the optional prefixes feature. So, the custom domains part feature doesn't exist (yet).

I'll be honest, I've never used the `Route::domain()` feature because I've never worked on a project that's needed it. So I'm a little bit hesitant of adding the custom domain feature to ShortURL myself because I might not do it right and might miss something off. But I'm more than happy to accept a PR if someone else wants to add this functionality though.

In the meantime, you might be able to use your custom domain by using a custom route (https://github.com/ash-jc-allen/short-url#custom-route). Do you think this could work for you? 🙂
